app/.eslintrc.js

```diff
@@ -0,0 +1,320 @@
+                },
+                "singleline": {
+                    "delimiter": "comma",
+                    "requireLast": false
+                }
+            }
+        ],
+        "@typescript-eslint/member-ordering": [
+            "error",
+            {
+                "classes": [
+                    "public-static-field",
+                    "public-instance-field",
+                    "private-static-field",
+                    "private-instance-field",
+                    "field",
+                    "constructor",
+                    "public-static-method",
+                    "public-instance-method",
+                    "private-static-method",
+                    "private-instance-method",
+                    "method"
+                ],
+                "interfaces": [
+                    "field",
+                    "constructor",
+                    "method",
+                    "signature"
+                ],
+                "typeLiterals": [
+                    "field",
+                    "constructor",
+                    "method",
+                    "signature"
+                ]
+            }
+        ],
+        "@typescript-eslint/method-signature-style": [
+            "error",
+            "method"
+        ],
+        "@typescript-eslint/naming-convention": [
+            "error",
+            {
+                "selector": "variableLike",
+                "format": ["camelCase"]
+            },
+            {
+                "selector": "memberLike",
+                "format": ["camelCase"]
+            },
+            {
+                "selector": "typeLike",
+                "format": ["PascalCase"]
+            },
+            {
+                "selector": "variable",
+                "format": ["camelCase", "UPPER_CASE"]
+            },
+            {
+                "selector": "parameter",
+                "format": ["camelCase"],
+                "leadingUnderscore": "allow"
+            },
+            {
+                "selector": "memberLike",
+                "modifiers": ["private"],
+                "format": ["camelCase"],
+                "leadingUnderscore": "allow"
+            },
+            {
+                "selector": "property",
+                "modifiers": ["static", "readonly"],
+                "types": ["boolean", "string", "number"],
+                "format": ["UPPER_CASE"]
+            },
+            {
+                "selector": "enumMember",
+                "format": ["UPPER_CASE"]
+            }
+        ],
+        "@typescript-eslint/no-base-to-string": "error",
+        "@typescript-eslint/no-confusing-non-null-assertion": "error",
+        "@typescript-eslint/no-dynamic-delete": "error",
+        "@typescript-eslint/no-invalid-void-type": "error",
+        "@typescript-eslint/no-parameter-properties": "error",
+        "@typescript-eslint/no-require-imports": "error",
+        "@typescript-eslint/no-throw-literal": "error",
```

Comment:
This breaks our way of throwing exceptions, could you offer a better way of doing it or remove this?

E.g. ContinentLockEvent, various throws there.

Reply:
Look into ApplicationException.ts. I made a small adjustment that resolves this.

Reply:
Ah gotcha